security/application.md

```diff
@@ -0,0 +1,412 @@
+every character even after you know the answer.
+
+## Encryption
+
+An encryption algorithm is one where a string can be made illegible and then
+returned back to the original string again, and where decrypting requires an
+out-of-band secret.
+
+Less abstractly: a string can be encrypted, and then to decrypt you must know
+the password.
+
+There are two kinds of encryption algorithms: symmetric and asymmetric.
+A symmetric algorithm is one where the same secret is used to encrypt it and
+decrypt it. An asymmetric algorithm is one where the string can be encrypted
+and decrypted using different secrets -- where the person encrypting cannot
+necessarily decrypt it.
+
+The most popular symmetric algorithms you'll encounter are AES and Twofish.
+These might be useful for encrypting a file to share with a group of people or
+for encrypting your filesystem. Rumor has it that 1Password uses something like
+it AES to encrypt an entire vault; it is encrypted at rest, and only decrypted
+when you enter the passphrase.
+
+Asymmetric encryption algorithms, also known as public/private keypair
+encryption, is more well-known -- in large part for how tricky they are to get
+right. Some famous ones are SSH, TLS (previously SSL), and PGP.  These start by
+generated a pair of encryption secrets known as the public and private keys.
+Anyone with the public key can encrypt a string, but only the holder of the
+private key can decrypt.
+
+(The math here is cool. I won't go into it.)
+
+In order for any of this to work, you need to get your hands on a confirmed
+public key. Each public key has a fingerprint -- an abbreviated and
+easily-confirmable portion of the entire secret. How this works in practice
+depends on the protocol.
+
+### SSH
+
+SSH defaults to a trust-on-first-use (TOFU) policy: the first time you connect
+to a server you are asked to confirm the server's public key fingerprint:
+
+```
+The authenticity of host heroku.com can't be established.
+RSA key fingerprint is 8tF0wX2WquK45aGKs/Bh1dKmBXH08vxUe0VCJJWOA/o.
+Are you sure you want to continue connecting (yes/no)?  
+```
+
+The server admin will need to tell you out of band whether that is the correct
+fingerprint ([Heroku publishes their fingerprint
+online](https://devcenter.heroku.com/articles/git-repository-ssh-fingerprints)).
+
+### PGP
+
+OpenPGP is a way for users to trust each other; therefore, fingerprint
+verification happens in person, often in a [keysigning
+party](http://mdcc.cx/gnupg/ksp_intro.en.html). People will exchange the
+fingerprint of their PGP key face-to-face, often written on paper, and then
+later will confirm that the key they have for the person matches the
+fingerprint on the paper.
+
+This mechanism is called a web of trust.
+
+### TLS
+
+Transport Layer Security is a way for a browser to trust a server. The browser
+ships with a list of trusted public keys. Each Web site serves up its own
+public key, plus a signature from another key. If the signature is by one of
```

Comment:
Maybe move the signing section up before the sections on key exchange, so that we've introduced signing when we get to here.